activerecord/lib/active_record/type/decimal.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Type
+    class Decimal < ActiveModel::Type::Decimal # :nodoc:
+      POSITIVE_INFINITY = BigDecimal("Infinity").freeze
+      NEGATIVE_INFINITY = BigDecimal("-Infinity").freeze
+
+      private
+
+        def cast_value(value)
+          casted_value = super
+          return 0 if casted_value == POSITIVE_INFINITY ||
```

Comment:
Early returns are for exceptional cases. When there are equal possibility of the values being any of the branches we should use `if/else`.

```ruby
if ....
  0
else
  casted_value
end
```